utils/check_doctest_list.py

```diff
@@ -54,7 +54,7 @@ def clean_doctest_list(doctest_file: str, overwrite: bool = False):
     all_paths = []
     with open(doctest_file, "r", encoding="utf-8") as f:
         for line in f:
-            line = line.strip()
+            line = line.strip().split(" ")[0]
```

Comment:
take into account the potential `comments` in some lines of `utils/not_doctested.txt`